### ISSUE TITLE
Proposal: Installing an additional executable for plugin's completion script

### ISSUE THREAD

**Issue by predatorray** [kubernetes-sigs/krew#812]
Hi there, I am the maintainer of a krew plugin. Kubectl recently released a new feature to support shell completion for plugins in its latest alpha release [v1.26.0-alpha.3](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.26.md#v1260-alpha3). It requires an additional executable named `kubectl_complete-<pluginName>` present on `$PATH`.

As discussed in the [pull request](https://github.com/kubernetes/kubernetes/pull/105867), `krew` does not provide a way to deliver this separate executable because `bin:` in the manifest yaml only expects one parameter. As a result, although I have implemented the completion script and managed to publish it to a homebrew tap simply by adding another `bin.install` (see: [predatorray/homebrew-brew#5](https://github.com/predatorray/homebrew-brew/pull/5/files#diff-752ae7b248a621ad81bc7efd279e2ed0e1450a562000df1539faf870c0709b24R16)), I am still not able to release this feature to the users using krew.

I am wondering do we have any plan to support this feature, for example, by supporting multiple parameters under `bin:` or providing another parameter called `completion`?

**Comment by ahmetb** [comment]
Yeah we should add support. I wasn't aware kubectl implemented support for it, as it was mostly in design discussion phase. It may take a while for us to get this done, so please be patient. Design proposals for this are appreciated.

**Comment by fredgate** [comment]
krew knows the name of the plugin. Couldn't it automatically copy the `kubectl_complete-pluginname` file into the PATH if there is one in the plugin package ?

**Comment by marckhouzam** [comment]
Until this is supported by krew and that plugins make use of it, people can use the following little project to generate the required completion script for about 50% of plugins installed by krew: https://github.com/marckhouzam/kubectl-plugin_completion
